FEXCore/Source/Utils/ArchHelpers/Arm64.cpp

```diff
@@ -1830,178 +1934,227 @@ HandleUnalignedAccess(FEXCore::Core::InternalThreadState* Thread, UnalignedHandl
+    }
+    std::atomic_ref<uint32_t>(PC[0]).store(STR, std::memory_order_release);
+    ClearICache(&PC[-1], 8);
+    // Back up one instruction and have another go
+    return std::make_pair(true, -4);
+  } else if ((Instr & RCPC2_MASK) == LDAPUR_INST) { // LDAPUR*
+    // Extract the 9-bit offset from the instruction
+    uint32_t LDUR = LDUR_INST;
+    LDUR |= Size << 30;
+    LDUR |= AddrReg << 5;
+    LDUR |= DataReg;
+    LDUR |= Instr & (0b1'1111'1111 << 9);
+    if (HandleType != UnalignedHandlerType::NonAtomic) {
+      ///< Ordering matters with cross-thread visibility!
+      std::atomic_ref<uint32_t>(PC[1]).store(DMB_LD, std::memory_order_release); // Back-patch the half-barrier.
+    }
+    std::atomic_ref<uint32_t>(PC[0]).store(LDUR, std::memory_order_release);
+    ClearICache(&PC[0], 8);
+    // With the instruction modified, now execute again.
+    return std::make_pair(true, 0);
+  } else if ((Instr & RCPC2_MASK) == STLUR_INST) { // STLUR*
+    uint32_t STUR = STUR_INST;
+    STUR |= Size << 30;
+    STUR |= AddrReg << 5;
+    STUR |= DataReg;
+    STUR |= Instr & (0b1'1111'1111 << 9);
+    if (HandleType != UnalignedHandlerType::NonAtomic) {
+      std::atomic_ref<uint32_t>(PC[-1]).store(DMB, std::memory_order_release); // Back-patch the half-barrier.
+    }
+    std::atomic_ref<uint32_t>(PC[0]).store(STUR, std::memory_order_release);
+
+    ClearICache(&PC[-1], 8);
+    // Back up one instruction and have another go
+    return std::make_pair(true, -4);
+  } else if ((Instr & ArchHelpers::Arm64::LDAXP_MASK) == ArchHelpers::Arm64::LDAXP_INST) { // LDAXP
+    /// This is handling the case of paranoid ARMv8.0-a atomic stores.
+    /// This backpatches the ldaxp+stlxp+cbnz if the previous `HandleCASPAL_ARMv8` didn't handle the case.
+    if (ArchHelpers::Arm64::HandleAtomicVectorStore(Instr, ProgramCounter)) {
+      return std::make_pair(true, 0);
+    } else {
+      LogMan::Msg::EFmt("Unhandled JIT SIGBUS LDAXP: PC: 0x{:x} Instruction: 0x{:08x}\n", ProgramCounter, PC[0]);
+      return NotHandled;
+    }
+  } else if ((Instr & ArchHelpers::Arm64::STLXP_MASK) == ArchHelpers::Arm64::STLXP_INST) { // STLXP
+    // Should not trigger - middle of an LDAXP/STAXP pair.
+    LogMan::Msg::EFmt("Unhandled JIT SIGBUS STLXP: PC: 0x{:x} Instruction: 0x{:08x}\n", ProgramCounter, PC[0]);
+    return NotHandled;
+  }
+
+  ///< Check if another thread backpatched this instruction before this thread got here
+  // Since we got here, this can happen in a couple situations:
+  // - Unhandled instruction (Shouldn't occur, programmer error)
+  // - Another thread backpatched an atomic access to be a non-atomic access
+  auto AtomicInst = std::atomic_ref<uint32_t>(PC[0]).load(std::memory_order_acquire);
+  if ((AtomicInst & LDSTREGISTER_MASK) == LDR_INST || (AtomicInst & LDSTUNSCALED_MASK) == LDUR_INST) {
+    ///< This atomic instruction likely was backpatched to a load.
```

Comment:
Why "likely"? When would it not be backpatched? What happens if we run this code  but it wasn't actually backpatched?

Reply:
Removed likely word. It will have been backpatched if we got there.